### PR TITLE
Fix flaky file_system_rules tests with better resource-matching.

### PR DESCRIPTION
**Pull request by yashtewari**
Resolves: https://github.com/elastic/security-team/issues/4484

### CODE REVIEW

**Comment by DaveSys911** [comment]
@yashtewari if it's a WIP, can we perhaps convert this to a draft?

File: tests/product/tests/test_file_system_rules.py [hunk 1]
```diff
@@ -34,13 +34,14 @@
...
-                      )),
-     *fs_tc.cis_1_1_19[3:],
+     *fs_tc.cis_1_1_19,
+    #  *skip_param_case(skip_list=fs_tc.cis_1_1_19[0:3],
+    #                   data_to_report=SkipReportData(
+    #                       url_title="security-team: #4484",
+    #                       url_link="https://github.com/elastic/security-team/issues/4484",
+    #                       skip_reason="known issue: flaky file_system_rules tests"
+    #                   )),
+    #  *fs_tc.cis_1_1_19[3:],
```
Comment (oren-zohar) [inline tests/product/tests/test_file_system_rules.py:44]: is it skipped? why are we commenting out this part?

File: tests/product/tests/test_file_system_rules.py [hunk 2]
```diff
@@ -86,8 +87,27 @@ def test_file_system_configuration(elastic_client,
                             param_value=param_value,
                             resource=resource)
 
-    def identifier(res):
-        return res.name in resource
+    def identifier(eval_resource):
+        if eval_resource.name not in resource:
+            return False
+
+        print('MATCH?', eval_resource)
```
Comment (oren-zohar) [inline tests/product/tests/test_file_system_rules.py:94]: leftover print?

**Review by oren-zohar — CHANGES_REQUESTED** [review]

**Review by oren-zohar — APPROVED** [review]
Amazing!

File: tests/product/tests/test_file_system_rules.py (hunk 1 above)
Comment (yashtewari) [reply tests/product/tests/test_file_system_rules.py]: PR wasn't ready.

File: tests/product/tests/test_file_system_rules.py (hunk 2 above)
Comment (yashtewari) [reply tests/product/tests/test_file_system_rules.py]: PR wasn't ready.

File: tests/product/tests/data/file_system/file_system_test_cases.py [hunk 3]
```diff
@@ -50,9 +50,9 @@
 
 cis_1_1_11 = [
     ('CIS 1.1.11', 'chmod', '0710', '/var/lib/etcd', 'failed'),
-    ('CIS 1.1.11', 'chmod', '0710', '/var/lib/etcd/some_file.txt', 'failed'),
+    # ('CIS 1.1.11', 'chmod', '0710', '/var/lib/etcd/some_file.txt', 'failed'),
     ('CIS 1.1.11', 'chmod', '0600', '/var/lib/etcd', 'passed'),
-    ('CIS 1.1.11', 'chmod', '0600', '/var/lib/etcd/some_file.txt', 'passed'),
+    # ('CIS 1.1.11', 'chmod', '0600', '/var/lib/etcd/some_file.txt', 'passed'),
 ]
```
Comment (gurevichdmitry) [inline tests/product/tests/data/file_system/file_system_test_cases.py:56]: @yashtewari, according to the CIS benchmark - "1.1.11 Ensure that the etcd data directory permissions are set to 700 or more restrictive (Automated)". So commented cases are not relevant and may be removed.

**Review by gurevichdmitry — APPROVED** [review]
@yashtewari LGTM.
nit, please remove redundant test cases for etcd dir checks.

File: tests/product/tests/data/file_system/file_system_test_cases.py (hunk 3 above)
Comment (yashtewari) [reply tests/product/tests/data/file_system/file_system_test_cases.py:56]: Yup, we don't actually need to evaluate the contents of that directory, so they were disabled. Removing as part of the upcoming merge.

**Pull request merged**